### ISSUE TITLE
pylint #5719 E0601 false negative

### ISSUE THREAD

**Issue by EgZvor** [PyCQA/pylint#1632]
I wonder how this was resolved https://www.logilab.org/ticket/5719, because it still doesn't work.
### Steps to reproduce
```
def f(n):
  if n == 0:
      if x == 1:
          pass
  else:
      print x
      x = 3
```

### Current behavior
nothing

### Expected behavior
e0601 raised

### pylint --version output
```
pylint 1.7.2, 
astroid 1.5.3
Python 2.7.13 (default, Jul 21 2017, 03:24:34) 
[GCC 7.1.1 20170630]
```

**Comment by PCManticore** [comment]
I can't reproduce this using the latest master. I'm getting this while running pylint over your code:
``E:  3, 9: Using variable 'x' before assignment (used-before-assignment)``
